docs/design-documents/design_template.md

```diff
@@ -0,0 +1,194 @@
+# Mbed-OS Design Document
+Write your feature title above, in this case its "Mbed-OS Design Document"
+
+# Table of Contents
+- [Mbed-OS Design Document](#mbed-os-design-document)
+- [Table of Contents](#table-of-contents)
+    - [Revision history](#revision-history)
+- [Introduction](#introduction)
+    - [Overview and Background](#overview-and-background)
+    - [Requirements and Assumptions](#requirements-and-assumptions)
+- [System Architecture and High-Level Design](#system-architecture-and-high-level-design)
+    - [System Architecture and Component Interaction](#system-architecture-and-component-interaction)
+- [Detailed Design](#detailed-design)
+- [Usage Scenarios and Examples](#usage-scenarios-and-examples)
+- [Tools and Configuration Changes](#tools-and-configuration-changes)
+- [Other Information](#other-information)
+    - [Reusability](#reusability)
+    - [Deprecations](#deprecations)
+    - [References](#references)
+    - [Custom Headings Here](#custom-headings-here)
+
+### Revision history
+1.0 - A brief description for this version, e.g. Initial revision - Author name - Date
```

Comment:
do we need revision here? It's in git so we see what has changed and when. No need to version this document like this

Reply:
@0xc0170 - I think we need the revision here because an author can make multiple changes in github to capture the new ideas or to match the implementation. But they all might point to one logical design. That's why I thought a revision here makes sense. For example, some one might push a change and then may decide to make some typo fixes/minor interface name changes which will be updated into the document. But it doesn't change the design logically. Its more like major design version and git versioning will be more like minor versioning.

Reply:
If we keep a revision number (as opposed to using git) we need to agree on what it should be. Maybe it's more appropriate to say: "This document's revision addresses Mbed OS 5.10 onwards"? That way it's more understandable to see what versions of the OS this document addresses.